src/Sarif.Viewer.VisualStudio.Core/RunningDocTableEventsHandler.cs

```diff
@@ -0,0 +1,158 @@
+        {
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterSave(uint docCookie)
+        {
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterAttributeChange(uint docCookie, uint grfAttribs)
+        {
+            return VSConstants.S_OK;
+        }
+
+        public int OnBeforeDocumentWindowShow(uint docCookie, int fFirstShow, IVsWindowFrame pFrame)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            if (ivsRunningDocTable != null)
+            {
+                IVsHierarchy hierarchy;
+                uint itemId;
+                IntPtr docData;
+                string fileName = null;
+
+                int hr = ivsRunningDocTable.GetDocumentInfo(docCookie, out uint a, out uint b, out uint c, out string pbstrMkDocument, out hierarchy, out itemId, out docData);
+                if (hr == VSConstants.S_OK && hierarchy != null)
+                {
+                    hierarchy.GetCanonicalName(itemId, out fileName);
+                    DocHandler_Service_Event(fileName);
+                }
+
+                // Now you have the file name of the document that was opened
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterDocumentWindowHide(uint docCookie, IVsWindowFrame pFrame)
+        {
+            return VSConstants.S_OK;
+        }
+
+        private void DocHandler_Service_Event(string filePath)
+        {
+            DocHandler_Service_Event(new FilesOpenedEventArgs() { FileOpened = new List<string>() { filePath } });
+        }
+
+        private void DocHandler_Service_Event(List<string> filePaths)
+        {
+            DocHandler_Service_Event(new FilesOpenedEventArgs() { FileOpened = filePaths });
+        }
+
+        private void DocHandler_Service_Event(FilesOpenedEventArgs e)
+        {
+            ServiceEvent?.Invoke(this, e);
+        }
+
+        /// <summary>
+        /// Called when the poll timer fires and sees if VS is done scanning the files in the workspace.
+        /// If so, the collection is updated and <see cref="FilesAdded"/> may be fired to report any new files.
+        /// For a solution:
+        /// We wait one poll period to ensure VS has fully loaded the solution. In practice, we've seen that VS
+        /// doesn't return any files for the solution if we ask for them right after the solution is opened.
+        /// For "Open Folder":
+        /// We wait until the file indexer service reaches a final state before we scan the workspace for files.
+        /// Depending on the size of the workspace we may have to restart the poll timer to keep checking the
+        /// file indexer service.
+        /// </summary>
+        /// <param name="state">State when the timer is fired.</param>
+#pragma warning disable VSTHRD100 // Avoid async void methods
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
This sounds worrisome.  Is this file a relative copy-paste from our repo?

Reply:
The poll timer needs it to be an async void, we do the same in our repo currently.